java/ql/lib/semmle/code/java/security/ArbitraryApkInstallationQuery.qll

```diff
@@ -0,0 +1,115 @@
+/** Provides dataflow configurations to reason about installation of arbitrary Android APKs. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.security.ArbitraryApkInstallation
+
+/**
+ * A dataflow configuration for flow from an external source of an APK to the
+ * `setData[AndType][AndNormalize]` method of an intent.
+ */
+private module ApkConf implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node node) { node instanceof ExternalApkSource }
+
+  predicate isSink(DataFlow::Node node) {
+    exists(MethodAccess ma |
+      ma.getMethod() instanceof SetDataMethod and
+      ma.getArgument(0) = node.asExpr() and
+      (
+        PackageArchiveMimeTypeConfiguration::hasFlowToExpr(ma.getQualifier())
+        or
+        InstallPackageActionConfiguration::hasFlowToExpr(ma.getQualifier())
+      )
+    )
+  }
+}
+
+module ApkConfiguration = DataFlow::Make<ApkConf>;
```

Comment:
Maybe change the name of the resulting module here - the result is not really a configuration, so maybe `ApkFlow` instead?